inst/include/Rcpp/macros/macros.h

```diff
@@ -29,9 +29,11 @@
 #ifndef BEGIN_RCPP
 #define BEGIN_RCPP                                                                               \
     int rcpp_output_type = 0 ;                                                                   \
+    int nprot = 0;                                                                               \
     (void)rcpp_output_type;                                                                      \
     SEXP rcpp_output_condition = R_NilValue ;                                                    \
     (void)rcpp_output_condition;                                                                 \
+    static SEXP stop_sym = Rf_install("stop");                                                   \
```

Comment:
Would be nicer if `stop` could be installed only once, instead of once per function, but that's an improvement I guess.

Reply:
@jjallaire would it make some sense to have something like : 

```cpp
SEXP stop_symbol() {
  static SEXP s = Rf_install("stop");
  return s;
}
```

somewhere at the top of the generated `RcppExports.cpp` so that the `stop` symbol is only installed once, instead of once per function ?

Reply:
You might also need to make this function static in order to support unloading DLLs on certain platforms. This is not important for packages but is important for `sourceCpp()`. IIRC, it caches dynlibs and those can be unloaded and reloaded several times in the same R session. On certain platforms, the ctors for the static variables failed to be called when the dynlib was loaded again, even though the dtors did get called on unloading, which caused crashes. I fixed this by giving static storage to the functions wrapping the static variables.